### PR TITLE
Add asset filter to pools list

### PR DESCRIPTION
**Pull request by ryangoree**
asset-filter

### CODE REVIEW

File: apps/hyperdrive-trading/src/ui/markets/PoolRow/PoolsList.tsx [hunk 1]
```diff
@@ -88,134 +106,295 @@ export function PoolsList(): ReactElement {
...
+): FiltersState {
+  switch (action.type) {
+    case "init":
+      const newState = {
+        assets: { ...state.assets },
+        chains: { ...state.chains },
+      };
+
+      for (const { hyperdrive, depositAssets } of action.pools) {
+        newState.chains[hyperdrive.chainId] ||= {
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/markets/PoolRow/PoolsList.tsx:351]: nit: `||=` is a neat (TIL!), but my preference for this is to just use an if-statement. It's more universally understood and keeps it readable for all levels.

```ts
if (!newState.chains[hyperdrive.chainId]) {
  newState.chains[hyperdrive.chainId] = { 
    id: hyperdrive.chainId,
    selected: false,
  };
}
```

**Review by DannyDelott — APPROVED** [review]
Nice!

File: apps/hyperdrive-trading/src/ui/markets/PoolRow/PoolsList.tsx (hunk 1 above)
Comment (ryangoree) [reply apps/hyperdrive-trading/src/ui/markets/PoolRow/PoolsList.tsx:351]: I understand the aversion, but the new language features are objectively simpler and like many new features in programming languages that make things easier, it's only a matter of time until it's ubiquitous. I think it's healthy for our engineering culture to embrace these new features the same way we have an optimistic welcoming view of innovations that make our lives easier.

I feel very strongly about this. Explicitly stating so in a style guide or something similar might help moving forward. I am going to merge, but am open to having an open conversation on this.

Comment (ryangoree) [reply apps/hyperdrive-trading/src/ui/markets/PoolRow/PoolsList.tsx:351]: They also give language engines a chance to optimize your code. A shorter syntax can do something more efficient under the hood while manually handling cases like this means the language is forces to evaluate each step of the logic.

**Pull request merged**